### PR TITLE
[SPARK-14988][PYTHON] SparkSession API follow-ups

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

Addresses comments in #12765.
## How was this patch tested?

Python tests.

### CODE REVIEW

**Comment by andrewor14** [comment]
@davies

File: python/pyspark/sql/conf.py [hunk 1]
```diff
@@ -33,62 +33,24 @@ def __init__(self, jconf):
...
-        True
-        """
-        iter = self._jconf.getOption(key).iterator()
-        if iter.hasNext():
-            return iter.next()
+        if default is None:
+            return self._jconf.get(key)
         else:
-            return None
+            return self._jconf.get(key, default)
```
Comment (davies) [inline python/pyspark/sql/conf.py]: Could you check the type for key and default, or you will get an weird py4j error?

File: python/pyspark/sql/tests.py [hunk 2]
```diff
@@ -1394,6 +1395,200 @@ def test_toDF_with_schema_string(self):
...
+        spark = self.sparkSession
+        spark.catalog._reset()
+        databases = [db.name for db in spark.catalog.listDatabases()]
+        self.assertEquals(databases, ["default"])
+        spark.sql("CREATE DATABASE some_db")
+        databases = [db.name for db in spark.catalog.listDatabases()]
+        self.assertEquals(sorted(databases), ["default", "some_db"])
+
+    def test_list_tables(self):
+        from catalog import Table
```
Comment (davies) [inline python/pyspark/sql/tests.py]: It's better to use full path
from pyspark.sql.catalog import Table

**Comment by andrewor14** [comment]
Merging into master. Thanks @davies.

**Pull request closed without merging**